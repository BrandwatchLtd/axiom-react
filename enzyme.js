// Prevent React 16 warnings on the console during tests using requestAnimationFrame
global.requestAnimationFrame = setTimeout;

import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });
