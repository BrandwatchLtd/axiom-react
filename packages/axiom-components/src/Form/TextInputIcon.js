import PropTypes from "prop-types";
import React, { Component } from "react";
import TextInputIconWrapper from "./TextInputIconWrapper";
import Icon from "../Icon/Icon";

export const TextInputIconRef = "TextInputIcon";

export default class TextInputIcon extends Component {
  static propTypes = {
    /** Alignment inside the input field */
    align: PropTypes.oneOf(["left", "right"]),
    /** Color of the TextInputIcon. */
    iconColor: PropTypes.string,
    /** Name of the icon. See <Icon>. */
    name: PropTypes.string.isRequired,
    /** Click handler, applies styling to indicate it is clickable. */
    onClick: PropTypes.func,
    /** Optional tooltip for the icon. */
    tooltip: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    /** Optional tooltip color. Default is "carbon" */
    tooltipColor: PropTypes.oneOf([
      "success",
      "warning",
      "error",
      "info",
      "carbon",
      "white",
    ]),
  };

  static typeRef = TextInputIconRef;

  render() {
    const {
      align,
      iconColor,
      name,
      onClick,
      tooltip,
      tooltipColor,
      ...rest
    } = this.props;

    return (
      <TextInputIconWrapper
        align={align}
        onClick={onClick}
        tooltip={tooltip}
        tooltipColor={tooltipColor}
        {...rest}
      >
        <Icon name={name} textColor={iconColor} size="1rem" />
      </TextInputIconWrapper>
    );
  }
}
