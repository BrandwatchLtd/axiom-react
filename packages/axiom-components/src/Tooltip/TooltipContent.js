import React, { Component } from 'react';
import ContextContent from '../Context/ContextContent';

export default class TooltipContent extends Component {
  render() {
    return (
      <ContextContent { ...this.props } padding="small" />
    );
  }
}
