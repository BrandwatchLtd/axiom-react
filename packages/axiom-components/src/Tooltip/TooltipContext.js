import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Context from '../Context/Context';

export default class TooltipContext extends Component {
  static propTypes = {
    color: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
  }
  render() {

    const {
      color,
      ...rest
    } = this.props;

    return (
      <Context color={ color } { ...rest } />
    );
  }
}
