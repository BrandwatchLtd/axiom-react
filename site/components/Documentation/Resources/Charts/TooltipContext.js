import React, { Component } from "react";
import {
  DropdownContext,
  DropdownContent,
  Paragraph,
} from "@brandwatch/axiom-components";

export default class TooltipContextExample extends Component {
  render() {
    return (
      <DropdownContext {...this.props} width="17rem">
        <DropdownContent>
          <Paragraph textColor="subtle">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis
            laoreet gravida mauris vel ultricies.
          </Paragraph>
        </DropdownContent>
      </DropdownContext>
    );
  }
}
