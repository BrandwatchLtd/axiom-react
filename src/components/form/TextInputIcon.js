import classnames from 'classnames';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Icon from '../icon/Icon';
import Link from '../typography/Link';

export const TextInputIconRef = 'TextInputIcon';

export default class TextInputIcon extends Component {
  static propTypes = {
    /** Alignment inside the input field */
    align: PropTypes.oneOf(['left', 'right']),
    /** Name of the icon. See <Icon>. */
    name: PropTypes.string.isRequired,
    /** Click handler, applies styling to indicate it is clickable. */
    onClick: PropTypes.func,
  };

  static defaultProps = {
    align: 'right',
  };

  static typeRef = TextInputIconRef;

  render () {
    const { align, name, onClick, ...rest } = this.props;
    const className = classnames('ax-input__icon', {
      [`ax-input__icon--align-${align}`]: align,
    });

    return (
      <span className={ className }>
        {
          onClick ? (
            <Link { ...rest } onClick={ onClick } style="subtle">
              <Icon name={ name } size="1rem" />
            </Link>
          ) : (
            <Icon name={ name } size="1rem" />
          )
        }
      </span>
    );
  }
}
