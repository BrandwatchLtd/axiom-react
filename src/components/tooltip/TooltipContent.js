import PropTypes from 'prop-types';
import { Component, cloneElement } from 'react';

export const TooltipContentRef = 'TooltipContent';

export default class TooltipContent extends Component {
  static propTypes = {
    children: PropTypes.node,
  };

  static typeRef = TooltipContentRef;

  render() {
    const { children, ...rest } = this.props;
    return cloneElement(children, rest);
  }
}
