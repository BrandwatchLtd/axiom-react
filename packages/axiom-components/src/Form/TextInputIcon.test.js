import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import TextInputIcon from "./TextInputIcon";
import TooltipContext from "../Tooltip/TooltipContext";
import TooltipContent from "../Tooltip/TooltipContent";

const getComponent = (props) => renderer.create(<TextInputIcon {...props} />);

describe("TextInputIcon", () => {
  let props;

  beforeEach(() => {
    props = {
      name: "twitter",
    };
  });

  it("renders with defaultProps", () => {
    const component = getComponent(props);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with align="left"', () => {
    props.align = "left";
    const component = getComponent(props);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with onClick", () => {
    props.onClick = () => {};
    const component = getComponent(props);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with the correct iconColor", () => {
    props.iconColor = "facebook";
    const component = getComponent(props);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
