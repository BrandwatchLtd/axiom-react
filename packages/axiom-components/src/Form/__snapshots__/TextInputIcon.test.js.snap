// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputIcon renders with defaultProps 1`] = `
<span
  className="ax-input__icon ax-input__icon--align-right"
>
  <svg
    className="ax-icon ax-icon--twitter"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<path d=\\"M0 13.526a6.586 6.586 0 0 0 4.86-1.36 3.285 3.285 0 0 1-3.067-2.28c.201.04.407.06.618.06.3 0 .693-.04.968-.115C1.878 9.529.85 8.203.85 6.613c0 .204.846.352 1.384.369C1.352 6.394.721 5.389.721 4.25c0-.601.136-1.166.418-1.65a9.29 9.29 0 0 0 6.754 3.429 3.287 3.287 0 0 1-.092-.747 3.28 3.28 0 0 1 5.674-2.246 6.554 6.554 0 0 0 2.084-.797 3.298 3.298 0 0 1-1.444 1.817A6.56 6.56 0 0 0 16 3.538a6.673 6.673 0 0 1-1.638 1.7c.007.14.01.281.01.424C14.371 10 11.07 15 5.031 15A9.301 9.301 0 0 1 0 13.526z\\" fill=\\"#1da1f2\\" fill-rule=\\"evenodd\\"/>",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</span>
`;

exports[`TextInputIcon renders with the correct iconColor 1`] = `
<span
  className="ax-input__icon ax-input__icon--align-right"
>
  <svg
    className="ax-icon ax-icon--twitter ax-text--color-facebook"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<path d=\\"M0 13.526a6.586 6.586 0 0 0 4.86-1.36 3.285 3.285 0 0 1-3.067-2.28c.201.04.407.06.618.06.3 0 .693-.04.968-.115C1.878 9.529.85 8.203.85 6.613c0 .204.846.352 1.384.369C1.352 6.394.721 5.389.721 4.25c0-.601.136-1.166.418-1.65a9.29 9.29 0 0 0 6.754 3.429 3.287 3.287 0 0 1-.092-.747 3.28 3.28 0 0 1 5.674-2.246 6.554 6.554 0 0 0 2.084-.797 3.298 3.298 0 0 1-1.444 1.817A6.56 6.56 0 0 0 16 3.538a6.673 6.673 0 0 1-1.638 1.7c.007.14.01.281.01.424C14.371 10 11.07 15 5.031 15A9.301 9.301 0 0 1 0 13.526z\\" fill=\\"#1da1f2\\" fill-rule=\\"evenodd\\"/>",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</span>
`;
